freqtrade/freqtradebot.py

```diff
@@ -334,9 +362,37 @@ def create_trade(self) -> bool:
             (buy, sell) = self.strategy.get_signal(_pair, interval, thistory)
 
             if buy and not sell:
+                experimental_check_depth_of_market = self.config.get('experimental', {}).\
+                    get('check_depth_of_market', {})
+                if (experimental_check_depth_of_market.get('enabled', False)) and\
+                        (experimental_check_depth_of_market.get('bids_to_ask_delta', 0) > 0):
+                    if self._check_depth_of_market_buy(_pair):
+                        return self.execute_buy(_pair, stake_amount)
+                    else:
+                        return False
                 return self.execute_buy(_pair, stake_amount)
         return False
 
+    def _check_depth_of_market_buy(self, pair: str) -> bool:
+        """
+        Checks depth of market before executing a buy
+        """
+        experimental_check_depth_of_market = self.config.get('experimental', {}).\
+            get('check_depth_of_market', {})
```

Comment:
maybe pass the `'check_depth_of_market'` dict from the parent function - the same line is there twice (L380 and L365) - and they are called 3 operations apart.

Reply:
I also agree. Thanks!